_episodes/03-dplyr.md

```diff
@@ -223,559 +223,439 @@ You can also nest functions (i.e. one function inside of another), like this:
-```
-
-The first few rows of the output are full of `NA`s, so if we wanted to remove
-those we could insert a `filter()` in the chain:
-
-
-```r
-surveys %>%
-  filter(!is.na(weight)) %>%
-  mutate(weight_kg = weight / 1000) %>%
-  head()
+iris %>%
+  mutate(Sepal.Ratio = Sepal.Length/Sepal.Width,
+         Petal.Ratio = Petal.Length/Petal.Width)
 ```
 
 ```
-## # A tibble: 6 x 14
-##   record_id month   day  year plot_id species_id sex   hindfoot_length
-##       <int> <int> <int> <int>   <int> <chr>      <chr>           <int>
-## 1       588     2    18  1978       2 NL         M                  NA
-## 2       845     5     6  1978       2 NL         M                  32
-## 3       990     6     9  1978       2 NL         M                  NA
-## 4      1164     8     5  1978       2 NL         M                  34
-## 5      1261     9     4  1978       2 NL         M                  32
-## 6      1453    11     5  1978       2 NL         M                  NA
-## # … with 6 more variables: weight <int>, genus <chr>, species <chr>,
-## #   taxa <chr>, plot_type <chr>, weight_kg <dbl>
+## # A tibble: 150 x 7
+##    Sepal.Length Sepal.Width Petal.Length Petal.Width Species Sepal.Ratio
+##           <dbl>       <dbl>        <dbl>       <dbl> <fct>         <dbl>
+##  1          5.1         3.5          1.4         0.2 setosa         1.46
+##  2          4.9         3            1.4         0.2 setosa         1.63
+##  3          4.7         3.2          1.3         0.2 setosa         1.47
+##  4          4.6         3.1          1.5         0.2 setosa         1.48
+##  5          5           3.6          1.4         0.2 setosa         1.39
+##  6          5.4         3.9          1.7         0.4 setosa         1.38
+##  7          4.6         3.4          1.4         0.3 setosa         1.35
+##  8          5           3.4          1.5         0.2 setosa         1.47
+##  9          4.4         2.9          1.4         0.2 setosa         1.52
+## 10          4.9         3.1          1.5         0.1 setosa         1.58
+## # … with 140 more rows, and 1 more variable: Petal.Ratio <dbl>
 ```
-
-`is.na()` is a function that determines whether something is an `NA`. The `!`
-symbol negates the result, so we're asking for every row where weight *is not* an `NA`.
+#removed the NA example because iris doesn't have NAs
 
 > ### Challenge {.challenge}
 >
->  Create a new data frame from the `surveys` data that meets the following
->  criteria: contains only the `species_id` column and a new column called
->  `hindfoot_half` containing values that are half the `hindfoot_length` values.
->  In this `hindfoot_half` column, there are no `NA`s and all values are less
->  than 30.
+>  Create a new data frame from the `plant_phys` data that meets the following
+>  criteria: contains only the `Species` column and a new column called
+>  `Cond_ratio` containing values that are `Cond_night` divided by the `Cond_day` values.
```

Comment:
Cool.  I think Kim was saying this ratio might be at the heart of one of their scientific questions.